tests/unit/test_client.py

```diff
@@ -2219,3 +2224,72 @@ def test_atlan_client_headers(client: AtlanClient):
         "x-atlan-client-origin": "product_sdk",
     }
     assert expected == client._session.headers
+
+
+def test_get_all_pagation(group_client, mock_api_caller):
+    mock_page_1 = [
+        {"id": "1", "alias": "Group3"},
+        {"id": "2", "alias": "Group4"},
+    ]
+    mock_api_caller._call_api.side_effect = [
+        {"records": mock_page_1},
+    ]
+
+    groups = group_client.get_all(limit=2)
+
+    assert len(groups) == 2
+    assert groups[0].id == "1"
+    assert groups[1].id == "2"
+    assert mock_api_caller._call_api.call_count == 1
+
+
+def test_get_all_empty_response_with_raw_records(group_client, mock_api_caller):
+    mock_api_caller.reset_mock()
+    mock_page_1 = []
+    mock_api_caller._call_api.side_effect = [
+        {"records": mock_page_1},
+    ]
+
+    groups = group_client.get_all()
+    assert len(groups) == 0
+
+
+def test_get_all_with_columns(group_client, mock_api_caller):
+    mock_api_caller.reset_mock()
+    mock_page_1 = [
+        {"id": "1", "alias": "Group1"},
+        {"id": "2", "alias": "Group2"},
+    ]
+    mock_api_caller._call_api.side_effect = [
+        {"records": mock_page_1},
+    ]
+
+    columns = ["alias"]
+    groups = group_client.get_all(limit=10, columns=columns)
+
+    assert len(groups) == 2
+    assert groups[0].id == "1"
+    assert groups[0].alias == "Group1"
+    mock_api_caller._call_api.assert_called_once()
+    query_params = mock_api_caller._call_api.call_args.kwargs["query_params"]
+    assert query_params["columns"] == columns
```

Comment:
I would place `mock_api_caller.reset_mock()` at the end of my unit test assertions instead of at the start.
```suggestion
    mock_page_1 = [
        {"id": "1", "alias": "Group1"},
        {"id": "2", "alias": "Group2"},
    ]
    mock_api_caller._call_api.side_effect = [
        {"records": mock_page_1},
    ]

    columns = ["alias"]
    groups = group_client.get_all(limit=10, columns=columns)

    assert len(groups) == 2
    assert groups[0].id == "1"
    assert groups[0].alias == "Group1"
    mock_api_caller._call_api.assert_called_once()
    query_params = mock_api_caller._call_api.call_args.kwargs["query_params"]
    assert query_params["columns"] == columns
    mock_api_caller.reset_mock()
```